components/product/ProductView/ProductView.tsx

```diff
@@ -19,15 +19,18 @@ interface Props {
 
 const ProductView: FC<Props> = ({ product }) => {
   const addItem = useAddItem()
-  const { price } = usePrice({
-    amount: product.price.value,
-    baseAmount: product.price.retailPrice,
-    currencyCode: product.price.currencyCode!,
-  })
   const { openSidebar } = useUI()
   const [loading, setLoading] = useState(false)
   const [choices, setChoices] = useState<SelectedOptions>({})
 
+  const variant = getVariant(product, choices)
+
+  const { price } = usePrice({
+    amount: variant?.price!.value || product.price.value,
+    baseAmount: variant?.price!.retailPrice || product.price.retailPrice,
+    currencyCode: variant?.price!.currencyCode! || product.price.currencyCode!,
```

Comment:
@royderks this fails at bigcommerce because expects variant price to not be null, so replace `variant?.price!.` with  `variant?.price?. `